modules/run-pex-as-resource/main.tf

```diff
@@ -33,16 +44,16 @@ resource "null_resource" "run_pex" {
   provisioner "local-exec" {
     when = destroy
     command = (
-      var.enable_destroy_provisioner
+      tobool(self.triggers.enable_destroy_provisioner)
       # NOTE: The nested string interpolation can not be extracted because of the reference to self.
-      ? "${local.python_call} ${var.destroy_command_args} ${var.pass_in_previous_triggers ? "${var.previous_trigger_option} '${jsonencode(self.triggers != null ? self.triggers : {})}'" : ""}"
+      ? "${self.triggers.python_call} ${self.triggers.destroy_command_args} ${tobool(self.triggers.pass_in_previous_triggers) ? "${self.triggers.previous_trigger_option} '${jsonencode(self.triggers != null ? self.triggers : {})}'" : ""}"
       : "echo 'Skipping delete provisioner'"
     )
     environment = merge(
       {
-        PYTHONPATH = module.pex_env.python_path
+        PYTHONPATH = self.triggers.python_path
       },
-      var.env,
+      jsondecode(self.triggers.env),
     )
```

Comment:
We should actually deprecate and remove this functionality, as using triggers has many gotchas as we discussed in https://github.com/gruntwork-io/package-terraform-utilities/pull/24

Specifically, this only works if the platform running `apply` and `destroy` are the same. The reason for that is that the data in the trigger is stored at `apply` time, so for example, if one runs apply in linux (e.g. from CI/CD), the python path and everything will be based on Linux. But then it will fail if you try to run `destroy` on Windows because it no longer refreshes the environment info (as it pulls straight from the stored snapshot in the triggers).

There's probably other gotchas we aren't thinking through. So given that we no longer have confidence we can make this in an intuitive and robust fashion, I think it is time to remove it. Instead, we should direct folks towards creating custom providers as an alternative.

I actually don't think anyone is really using this anyway (I believe this is only used in EKS module here https://github.com/gruntwork-io/terraform-aws-eks/blob/master/modules/eks-cluster-control-plane/main.tf#L164).

It's a shame because this was a nice way to have a super light weight custom provider...

Reply:
Are you saying to remove the destroy-time provisioner? Or to remove the entire `run-pex-as-resoource` module?

Reply:
Sorry I should have clarified. I am suggesting we remove the destroy time provisioner feature, but keep the `run-pex-as-resource` module as that is generally useful (e.g. the ECS deployment checker) and still supported by tf13.

Reply:
Fair enough! Destroy-time provisioner removed: https://github.com/gruntwork-io/package-terraform-utilities/pull/30/commits/29a4ac2b201e3912e0aadaa205bed8691ccac9b0.

Reply:
And some more stuff to clean up: https://github.com/gruntwork-io/package-terraform-utilities/pull/30/commits/525c41ad74eb7b80d6152f74b3a10cce26ca3c31 and https://github.com/gruntwork-io/package-terraform-utilities/pull/30/commits/3927958517f28f17d4e47f7aef20aa84cf677f56.